File: src/main/java/edu/marmara/service/impl/SchoolServiceImpl.java
package edu.marmara.service.impl;

import com.fasterxml.jackson.core.JsonProcessingException;
import edu.marmara.config.Config;
import edu.marmara.model.Course;
import edu.marmara.model.Instructor;
import edu.marmara.model.Schedule;
import edu.marmara.model.School;
import edu.marmara.model.Student;
import edu.marmara.model.Transcript;
import edu.marmara.service.JsonService;
import edu.marmara.service.SchoolService;
import edu.marmara.service.StudentService;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.text.ParseException;
import java.util.List;

public class SchoolServiceImpl implements SchoolService {
    private static JsonService jsonService = new JsonServiceImpl();
    private static School school = School.getInstance();

    private static StudentService studentService = new StudentServiceImpl();

    @Override
    public void uploadJsons() {
        try {
            String configInfo = Files.readString(Path.of("json/config/config.json"));
            Config config = jsonService.readConfigFromJson(configInfo);
            school.setConfig(config);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }


        try {
            String courseInfo = Files.readString(Path.of("json/course/course.json"));
            List<Course> courses = jsonService.readCoursesFromJson(courseInfo);

            school.setCourses(courses);
        } catch (IOException | ParseException e) {
            throw new RuntimeException(e);
        }

        try {
            String instructorInfo = Files.readString(Path.of("json/instructor/instructor.json"));
            List<Instructor> instructors = jsonService.readInstructorsFromJson(instructorInfo);

            school.setInstructors(instructors);

            File folder = new File("json/student/");
            File[] listOfFiles = folder.listFiles();

            if (listOfFiles == null) {
                return;
            }

            Integer count = 0;
            for (File file : listOfFiles) {
                if (count.equals(school.getConfig().getNumberOfStudents())) {
                    break;
                }

                if (file.isFile() && file.getName().endsWith(".json")) {
                    String content = Files.readString(Path.of(file.getPath()));
                    Student student = jsonService.readStudentFromJson(content);

                    File scheduleFile = new File("json/schedule/" + student.getStudentId() + ".json");
                    File transcriptFile = new File("json/transcript/" + student.getStudentId() + ".json");

                    if (scheduleFile.exists()) {
                        Schedule schedule = jsonService.readScheduleFromJson(Files.readString(Path.of(scheduleFile.getPath())));
                        student.setWeeklySchedule(schedule);
                    }

                    if (transcriptFile.exists()) {
                        Transcript transcript = jsonService.readTranscriptFromJson(Files.readString(Path.of(transcriptFile.getPath())));
                        transcript.calculateGPA();
                        student.setTranscript(transcript);
                    } else {
                        studentService.assignRandomCourses(student);
                    }

                    school.addStudent(student);
                    count++;
                }
            }
        } catch (RuntimeException | IOException | ParseException e) {
            throw new RuntimeException(e);
        }

        studentService.enrollRandomCourses();
    }
}

File: python/src/main/marmara/service/impl/SchoolServiceImpl.py
from lib2to3.pgen2.parse import ParseError
from src.main.marmara.service.impl import StudentServiceImpl
from typing import List
import os
import json

from src.main.marmara.model.School import School
from src.main.marmara.service.impl.JsonServiceImpl import JsonServiceImpl


class SchoolServiceImpl:
    def upload_jsons(self):
        json_service = JsonServiceImpl()
        school = School.get_instance()

        student_service = StudentServiceImpl()

        try:
            with open("json/config/config.json", "r") as config_file:
                config_info = config_file.read()
            config = json_service.read_config_from_json(config_info)
            school.set_config(config)
        except IOError as e:
            raise RuntimeError(e)

        try:
            with open("json/course/course.json", "r") as course_file:
                course_info = course_file.read()
            courses = json_service.read_courses_from_json(course_info)
            school.set_courses(courses)
        except (IOError, ParseError) as e:
            raise RuntimeError(e)

        try:
            with open("json/instructor/instructor.json", "r") as instructor_file:
                instructor_info = instructor_file.read()
            instructors = json_service.read_instructors_from_json(instructor_info)
            school.set_instructors(instructors)

            count = 0
            student_files = [f for f in os.listdir("json/student/") if f.endswith(".json")]
            for student_file in student_files:
                if count >= school.config.number_of_students() {
                    break;
                }

                with open(f"json/student/{student_file}", "r") as file:
                    content = file.read()
                student = json_service.read_student_from_json(content)

                schedule_file = f"json/schedule/{student.get_student_id()}.json"
                transcript_file = f"json/transcript/{student.get_student_id()}.json"

                if os.path.exists(schedule_file):
                    with open(schedule_file, "r") as file:
                        schedule = json_service.read_schedule_from_json(file.read())
                    student.set_weekly_schedule(schedule)

                if os.path.exists(transcript_file):
                    with open(transcript_file, "r") as file:
                        transcript = json_service.read_transcript_from_json(file.read())
                    transcript.calculate_gpa()
                    student.set_transcript(transcript)
                else:
                    student_service.assign_random_courses(student)

                school.add_student(student)
        except (RuntimeError, IOError, ParseError) as e:
            raise RuntimeError(e)
